base/deprecated.jl

```diff
@@ -1499,4 +1499,12 @@ end
 # Calling promote_op is likely a bad idea, so deprecate its convenience wrapper promote_eltype_op
 @deprecate promote_eltype_op(op, As...) promote_op(op, map(eltype, As)...)
 
+# These conversions should not be defined, see #19896
+Base.convert{T<:Number}(::Type{T},x::Dates.Period) = error("conversion between Periods and Numbers is not defined")
+Base.convert{T<:Dates.Period}(::Type{T},x::Real) = error("conversion between Periods and Numbers is not defined")
+@deprecate convert{R<:Real}(::Type{R},x::Dates.DateTime) R(Dates.datetime2rata(x))
+@deprecate convert{R<:Real}(::Type{R},x::Dates.Date)     R(Dates.datetime2rata(x))
+@deprecate convert(::Type{Dates.DateTime}, x::Real) Dates.rata2datetime(x)
+@deprecate convert(::Type{Dates.Date}, x::Real)     Dates.Date(Dates.rata2datetime(x))
```

Comment:
I would suggest that we add the following methods:

```julia
Base.convert(::Type{DateTime}, x::Millisecond) = DateTime(Dates.UTInstant(x))  # Converts Rata Die milliseconds to a DateTime
Base.convert(::Type{Date}, x::Day) = Date(Dates.UTInstant(x))  # Converts Rata Die days to a Date
```

That way we can have a reasonable way of converting Reals into DateTimes and Dates:
```julia
julia> convert(Date, Day(730120))
2000-01-01

julia> convert(DateTime, Millisecond(63082368000000))
2000-01-01T00:00:00
```